driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -5,23 +5,80 @@ use dfusion_core::models;
 
 use chrono::Utc;
 use log::{debug, error};
-use serde::Deserialize;
-use serde_json::{json, Value};
 use std::collections::{HashMap, HashSet};
 use std::fs::File;
-use std::io::BufReader;
+use std::io::{BufReader, BufWriter, Read, Write};
 use std::process::Command;
 use web3::types::H160;
 
 const RESULT_FOLDER: &str = "./results/tmp/";
 
 type PriceMap = HashMap<u16, u128>;
 
+mod solver_output {
+    use serde::Deserialize;
+    use std::collections::HashMap;
+    use std::vec::Vec;
+
+    #[derive(Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Order {
+        pub exec_sell_amount: Option<String>,
+        pub exec_buy_amount: Option<String>,
+    }
+
+    #[derive(Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Output {
+        pub orders: Vec<Order>,
+        pub prices: HashMap<String, Option<String>>,
+    }
+}
+
+mod solver_input {
+    use serde::Serialize;
+    use std::collections::HashMap;
+    use std::vec::Vec;
+
+    #[derive(Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Order {
+        #[serde(rename = "accountID")]
+        pub account_id: String,
+        pub sell_token: String,
+        pub buy_token: String,
+        pub sell_amount: String,
+        pub buy_amount: String,
+        #[serde(rename = "ID")]
+        pub id: String,
+    }
+
+    #[derive(Serialize)]
+    pub struct Fee {
+        pub token: String,
+        pub ratio: f64,
+    }
+
+    pub type Accounts = HashMap<String, HashMap<String, String>>;
+
+    #[derive(Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Input {
+        pub tokens: Vec<String>,
+        pub ref_token: String,
+        pub accounts: Accounts,
+        pub orders: Vec<Order>,
+        // TODO vk: is this needed even though its never set to anything?
```

Comment:
This would be a question for `dex-solver` which would make use of them to estimate prices if they were provided. @twalth3r might know if this is still needed.

Reply:
We currently don't use previous prices, because we currently don't get any price info from the services. Not sure if this will always stay the case though..

Reply:
Right, so in this case, we should leave it possible as a TODO - for implementation in the future.

Reply:
@twalth3r can we completely omit the field when it's empty or does the solver expect it to be there (even when empty)

Reply:
Currently it is expected, but it would necessary to change 3 lines to make it optional - should I do this?

Reply:
Yes please.

Reply:
Change has been merged: https://github.com/gnosis/dex-solver/pull/91